### PR TITLE
License cleanup part 2

### PR DESCRIPTION
**Pull request by matthiasblaesing**
@emilianbold another set of cleanups for licenses/build issues. The individual commits hold
more information. The core info:

- maven-indexer was released as 6.0.0 and already contains the patch, so it can be dropped and we can switch to the maven artifact. The notice file was missing
- BIndex contains MIT licensed code and so needs an enhanced license and also notice info, which was extracted.

As before If not vetoed, I would cherry-pick this directly into release90 after review for master.

### CODE REVIEW

File: maven.indexer/external/indexer-core-6.0.0-notice.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Maven :: Indexer Core
+Copyright 2002-2017 The Apache Software Foundation
+
+This product includes software developed at
```
Comment (emilianbold) [inline maven.indexer/external/indexer-core-6.0.0-notice.txt]: I believe we can remove these two lines?

File: apisupport.harness/external/bindex-2.2-notice.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+OSGi OBR BIndex Program
+
+OSGi™ is a trademark, registered trademark, or service mark of The OSGi Alliance
+in the US and other countries. Java™ is a trademark, registered trademark, or
```
Comment (emilianbold) [inline apisupport.harness/external/bindex-2.2-notice.txt]: So sad when the notice is ancient...

File: maven.indexer/external/indexer-core-6.0.0-notice.txt (hunk 1 above)
Comment (matthiasblaesing) [reply maven.indexer/external/indexer-core-6.0.0-notice.txt]: We could, but we don't need to do - the ant task, that merges the notice files, already removes the "This product includes ..." lines. So I took the whole notice file.

Comment (emilianbold) [reply maven.indexer/external/indexer-core-6.0.0-notice.txt]: OK.

**Comment by matthiasblaesing** [comment]
I pushed three more commits into this branch (only new commits):
- the spring jars indeed hold a notice, this is now present
- testng indeed contains MIT licensed jquery and jcommander holds a NOTICE file (if you dig in the repository, in the distribution it is missing)
- winp might be discussable. Instead of patching winp, I switched to a maintained version (https://github.com/kohsuke/winp). The libraries code today is very similar to the state after patching (modulo a few years of fixes), so I think this is save. The API is stable (checked with ``japi-compliance-checker`). Part of the winp problem was, that the license integration code is not prepared for files showing up at build time (winp was build from source) and not coming from the normal download route.

**Comment by emilianbold** [comment]
> testng indeed contains MIT licensed jquery and jcommander holds a NOTICE file (if you dig in the repository, in the distribution it is missing)

It feels like we are somehow going above and beyond our duty if we not only inspect the 3rd party release but look into the repository, etc.

> winp might be discussable. Instead of patching winp, I switched to a maintained version (https://github.com/kohsuke/winp). The libraries code today is very similar to the state after patching (modulo a few years of fixes), so I think this is save. The API is stable (checked with ``japi-compliance-checker`). 

It would be nice to have this a separate PR.

File: libs.springframework/external/spring-framework-4.0.1.RELEASE-notice.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Spring Framework 4.0.1.RELEASE
+Copyright (c) 2002-2014 Pivotal, Inc.
+
+This product is licensed to you under the Apache License, Version 2.0
+(the "License"). You may not use this product except in compliance with
+the License.
+
+This product may include a number of subcomponents with separate
```
Comment (emilianbold) [inline libs.springframework/external/spring-framework-4.0.1.RELEASE-notice.txt]: This blurb seems pointless to add in the NOTICE file. The whole point of the NOTICE (and LICENSE) file is to be exhaustive, so there is nothing that 'may' be included...

Comment (matthiasblaesing) [reply libs.springframework/external/spring-framework-4.0.1.RELEASE-notice.txt]: Agreed. I did this in the past and just overlooked, that the second and third paragraph are superfluous. This just leaves the copyright info.

**Pull request merged**